bug fix for losing shape on wrapper tensor for DTensor

teja-rao opened the pull request:
Summary: Wrapper tensor for DTensor is losing shape in offload_tensor. This PR fixes this bug.

Test Plan:
updated the test. Test fails with old code and passes with the fix.

Rollback Plan:

Differential Revision: D77269733




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

mikaylagawarecki commented on torch/distributed/checkpoint/_state_dict_stager.py:
```diff
@@ -183,7 +183,7 @@ def _offload_tensor(self, x, memo, non_blocking=False):
         """
         # Create a new empty tensor on CPU
-        y = x.new_empty([], device="cpu")
+        y = x.new_empty(x.size(), device="cpu")
```
nit: maybe only for wrapper subclasses to avoid allocating memory otherwise

teja-rao committed:
bug fix for losing shape on wrapper tensor for DTensor (#156774)

Summary:

Wrapper tensor for DTensor is losing shape in offload_tensor. This PR fixes this bug.

Test Plan:
updated the test. Test fails with old code and passes with the fix.

Rollback Plan:

Reviewed By: mikaylagawarecki, Saiteja64

Differential Revision: D77269733
